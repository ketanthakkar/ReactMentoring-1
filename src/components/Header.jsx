import React from 'react';
import {Route, Switch} from 'react-router-dom';
import styled from 'styled-components';

import Logo from './Logo';
import SearchField from './SearchField';
import InfoBar from './InfoBar';
import Details from './Details';

import background from '../img/background.jpg';

const HeaderSection = styled.div`
    width: 100%;
    height: 100%;
    background-image: linear-gradient(rgba(0, 0, 0, 0.8), rgba(0, 0, 0, 0.4)), url(${background});
`;

const HeaderWrapper = styled.div`
    width: 900px;
    margin: 0 auto;
    padding-top: 20px;
`;

export class Header extends React.Component {
    render() {
        return (
            <HeaderSection>
                <HeaderWrapper>
                    <Logo/>
                    <Switch>
                        <Route path="/film/:id" component={Details}/>
                        <Route path="/" component={SearchField}/>
                    </Switch>
                </HeaderWrapper>
                <InfoBar/>
            </HeaderSection>
        )
    }
}

export default Header;
