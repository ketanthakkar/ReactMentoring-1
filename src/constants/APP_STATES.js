const APP_STATES = {
    SEARCH_PAGE: 'search page',
    DETAILS_PAGE: 'details page'
};

export default APP_STATES;