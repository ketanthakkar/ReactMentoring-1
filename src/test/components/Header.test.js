import React from 'react';
import { shallow } from 'enzyme';

import Header from 'components/Header';

import APP_STATES from 'constants/APP_STATES';

describe('Header', () => {
    const header = shallow(<Header />);

    it('should be like snapshot when AppState is SearchPage', () => {
        header.setProps({appState: APP_STATES.SEARCH_PAGE});
        expect(header).toMatchSnapshot();
    });

    it('should be like snapshot when AppState is DetailsPage', () => {
        header.setProps({appState: APP_STATES.DETAILS_PAGE});
        expect(header).toMatchSnapshot();
    });
});