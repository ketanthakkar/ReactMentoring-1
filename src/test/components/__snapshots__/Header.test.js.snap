// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should be like snapshot when AppState is DetailsPage 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    appState="details page"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Header__HeaderWrapper>
          <Logo />
          <Switch>
            <Route
              component={[Function]}
              path="/film/:id"
            />
            <Route
              component={[Function]}
              path="/"
            />
          </Switch>
        </Header__HeaderWrapper>,
        <Connect(InfoBar) />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Logo />,
            <Switch>
              <Route
                component={[Function]}
                path="/film/:id"
              />
              <Route
                component={[Function]}
                path="/"
              />
            </Switch>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  path="/film/:id"
                />,
                <Route
                  component={[Function]}
                  path="/"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/film/:id",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Header__HeaderWrapper>
            <Logo />
            <Switch>
              <Route
                component={[Function]}
                path="/film/:id"
              />
              <Route
                component={[Function]}
                path="/"
              />
            </Switch>
          </Header__HeaderWrapper>,
          <Connect(InfoBar) />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Logo />,
              <Switch>
                <Route
                  component={[Function]}
                  path="/film/:id"
                />
                <Route
                  component={[Function]}
                  path="/"
                />
              </Switch>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Route
                    component={[Function]}
                    path="/film/:id"
                  />,
                  <Route
                    component={[Function]}
                    path="/"
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "path": "/film/:id",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "path": "/",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Header should be like snapshot when AppState is SearchPage 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    appState="search page"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Header__HeaderWrapper>
          <Logo />
          <Switch>
            <Route
              component={[Function]}
              path="/film/:id"
            />
            <Route
              component={[Function]}
              path="/"
            />
          </Switch>
        </Header__HeaderWrapper>,
        <Connect(InfoBar) />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Logo />,
            <Switch>
              <Route
                component={[Function]}
                path="/film/:id"
              />
              <Route
                component={[Function]}
                path="/"
              />
            </Switch>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  path="/film/:id"
                />,
                <Route
                  component={[Function]}
                  path="/"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/film/:id",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Header__HeaderWrapper>
            <Logo />
            <Switch>
              <Route
                component={[Function]}
                path="/film/:id"
              />
              <Route
                component={[Function]}
                path="/"
              />
            </Switch>
          </Header__HeaderWrapper>,
          <Connect(InfoBar) />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Logo />,
              <Switch>
                <Route
                  component={[Function]}
                  path="/film/:id"
                />
                <Route
                  component={[Function]}
                  path="/"
                />
              </Switch>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Route
                    component={[Function]}
                    path="/film/:id"
                  />,
                  <Route
                    component={[Function]}
                    path="/"
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "path": "/film/:id",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "path": "/",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
